tests/browser/index.test.js

```diff
@@ -310,6 +310,35 @@ suite('<Entity events/>', () => {
     });
   });
 
+  test('can remove event listener when component is detached', function (done) {
+    let bazHandlerCalled = false;
+    function bazHandler () { bazHandlerCalled = true }
+
+    ReactDOM.render(
+      <Scene>
+        <Entity id='foo'>
+          <Entity id='bar' events={{baz: bazHandler}}/>
+        </Entity>
+      </Scene>,
+      div
+    );
+    div.querySelector('a-scene').addEventListener('loaded', () => {
+      ReactDOM.render(
+        <Scene>
+          <Entity id='foo'/>
+        </Scene>,
+        div
+      );
+      div.querySelector('a-entity').emit('baz');
```

Comment:
Without your changes, the handler will get called? I'm not sure how the handler can get called if the event is not emitted directly on the entity with the event listeners. Because the we're calling it on the parent entity. Was it propagating down to the child somehow?